src/core/document.js

```diff
@@ -261,6 +262,55 @@ class Page {
     );
   }
 
+  async saveNewAnnotations(handler, task, annotations) {
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: this.pageIndex,
+      idFactory: this._localIdFactory,
+      fontCache: this.fontCache,
+      builtInCMapCache: this.builtInCMapCache,
+      standardFontDataCache: this.standardFontDataCache,
+      globalImageCache: this.globalImageCache,
+      options: this.evaluatorOptions,
+    });
+
+    const pageDict = this.pageDict;
```

Comment:
For XFA-documents this will be the empty dictionary, which (obviously) cannot be modified and generally I'm not sure if it even makes sense to try and support adding annotations in XFA-documents.

So, at least for now, you may want to simply throw at the top of this method if `this.xfaFactory` is defined?